sdk/identity/azure-identity/test/ut/token_credential_impl_test.cpp

```diff
@@ -340,72 +340,6 @@ TEST(TokenCredentialImpl, NoToken)
       }));
 }
 
-TEST(TokenCredentialImpl, CurrentJsonParserQuirksAndLimitations)
```

Comment:
Should we keep this test?

What this is testing seems to be useful, regardless of the JSON parser used in the implementation.

It isn't directly testing the helpers like `GetPropertyValueAsInt64`, so I would expect it to pass even when the impl now uses nlohmann.

Reply:
Other tests cover parsing ints, and all other things, they just are not being touched in this PR, because they all pass as expected. But they have sufficient coverage.
This test is not needed anymore. Previous implementation would successfully parse a string that starts with double quote and "ends with" a single quote, and these quirks were being tested. Now that the implementation is updated, this test has started to fail as expected, and it is no longer needed.